cli/config.go

```diff
@@ -0,0 +1,69 @@
+package cli
+
+import (
+	"strconv"
+
+	"github.com/dominikbraun/timetrace/core"
+	"github.com/dominikbraun/timetrace/out"
+	"github.com/spf13/cobra"
+)
+
+func configCommand(t *core.Timetrace) *cobra.Command {
+	configCmd := &cobra.Command{
+		Use:   "config",
+		Short: "Manage configs",
+		Run: func(cmd *cobra.Command, args []string) {
+			_ = cmd.Help()
+		},
+	}
+
+	configCmd.AddCommand(setConfigCommand(t))
+	configCmd.AddCommand(getConfigCommand(t))
+	return configCmd
+}
+
+func setConfigCommand(t *core.Timetrace) *cobra.Command {
+	config := &cobra.Command{
+		Use:   "set <KEY> <VALUE>",
+		Short: "Set config value",
+		Args:  cobra.ExactArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+
+		},
+	}
+	return config
+}
+
+func getConfigCommand(t *core.Timetrace) *cobra.Command {
+	config := &cobra.Command{
+		Use:   "get <KEY>",
+		Short: "Get config values",
+		Args:  cobra.MaximumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			configKeys := make([][]string, 4)
+			configKeys[0] = createLine("store", t.Config().Store)
+			configKeys[1] = createLine("editor", t.Config().Editor)
+			configKeys[2] = createLine("reportPath", t.Config().ReportPath)
+			configKeys[3] = createLine("use12Hours", strconv.FormatBool(t.Config().Use12Hours))
+			if len(args) == 0 {
+
+				out.Table([]string{"Key", "Value"}, configKeys, nil)
+				return
+			}
+			for _, configKey := range configKeys {
+				if configKey[0] == args[0] {
+					slice := configKeys[0:1]
+					out.Table([]string{"Key", "Value"}, slice, nil)
+				}
+			}
+		},
+	}
+	return config
+}
+
+func createLine(key string, value string) []string {
+	line := make([]string, 2)
```

Comment:
JFYI: I am not sure if you know it, but you don't have to use `make` to create slices in all cases.
This are also possible options:
```go
// This case is very compact, so you may not neede the createLine as you can inline that directly...
return []string{key, value}
```

You can also just append the values:  
Interesting aspect about this: If a value is of type `[]string`, but it is not set yet (e.g. nil), you can still just use append and Go will handle it. (This is only true for slices, not for maps. These have to be created with make always)
```go
func createLine(key string, value string) []string {
	var line []string
	line = append(line, key)
	line = append(line, value)
	return line
}
```

Or append both at once:
```go
func createLine(key string, value string) []string {
	var line []string
	line = append(line, key, value)
	return line
}
```

The last two examples have the downside that Go may have to grow the slice (not likely in this case as its only two values...)  
It basically works like ArrayList in java.

Reply:
@aligator thx for the help.